GOVERNANCE.md

```diff
@@ -131,3 +122,75 @@ the voting process should be considered.
+* **Approvers** are able to both review and approve code contributions. While code review is focused on code quality and correctness, approval is focused on holistic acceptance of a contribution.  Approvers can merge changes.
+
+Note that neither of these roles have voting powers in conflict resolution, these roles are for the code base change approval process only.
+
+### Pull Request Flow
+
+The general flow for a pull request approval process is as follows:
+
+1. Author submits the pull request
+1. Reviewers and approvers for the applicable code areas review the pull request and provide feedback that the author integrates
+1. Reviewers and/or approvers signify their LGTM on the pull request
+1. An approver approves the pull request based on at least one LGTM from the previous step
+    1. Note that the approver can heavily lean on the reviewer for examining the pull request at a finely grained detailed level. The reviewers are trusted members and approvers can leverage their efforts to reduce their own review burden.
+1. An approver merges the pull request into the target branch (master, release, etc.)
+
+### Role Assignments
+
+#### Declarations
+
+All roles will be assigned by the usage of [`CODE-OWNERS`](CODE-OWNERS) files committed to the code base.
+These assignments will be initially be defined in a single file at the root of the repo and it will describe all assigned roles for the entire code base.
+As we incorporate automation (i.e. bots) into this change acceptance process in the future, we can reorganize this initial single owners file into separate files amongst the codebase as the automation necessitates.
+
+The format of the file can start with simply listing the reviewers and approvers for areas of the code base using a YAML format:
+
+```yaml
+areas:
+  feature-foo:
+    approvers:
+    - alice
+    - bob
+    reviewers:
+    - carol
+```
+
+#### Process
+
+All new reviewers and approvers must be nominated by someone (anyone) opening a pull request that adds the nominated person’s name to the appropriate [`CODE-OWNERS`](CODE-OWNERS) files in the appropriate roles.
+The maintainer team will approve this nomination in the standard voting and conflict resolution process.
+
+Note that new reviewer/approver nominations do not go through the standard pull request approval process described previously, only the maintainer team can approve additions of members to the change approval roles.
+
+### Permissions
+
+Role assignees will be made part of the following Rook organization teams with the given permissions:
+
+* **Reviewers:** added to Contributors team so they have write permissions to the repo to assign issues, add labels to issues, add issues to milestones and projects, etc. but cannot merge to protected branches such as `master` and `release-*`.
+* **Approvers:** added to the Contributors team and additionally added to the new Approvers team that will have access to merge to protected branches.
+
+### Automation
```

Comment:
i think tracking the idea of automation should be done in an issue, I will do this right after the PR is merged so that I can cleanly reference this doc section from the issue ticket.